### PR TITLE
Con 2763 20221208 update cvc workflow

### PR DESCRIPTION
**Pull request by nbutler77**
Updated cvc workflow:

 - Staging images is now done using `cvc stage-image`
 - Images are uploaded using `cvc upload-images`, which also preferentially adds new frame data to `index.json`
 - Adds new `unstage-image` command, which will unstage an image, given its full path
 - JSONL is now validated "in-place", rather than concatenated into a single `index.json` file.

**File Changes:**

`.gitignore`
 - Now ignores `.vscode/` directory

`FLIR/conservator/cli/cvc.py`
 - Renamed commands, added new `unstage-image` command

`FLIR/conservator/jsonl_to_index_json.py`
 - Removed module, no longer needed

`FLIR/conservator/local_dataset.py`
 - Refactored code to prefer using `index.json` rather than jsonl files; added method to check for error in `index.json`

`docs/usage/cvc_guide.rst`
 - Updated docs to reflect updates

`examples/create_dataset_frames.py`
 - Removed `x-amz-meta-originalfilename` field in upload, no longer used (and actually causes an error now)

`test/integration/test_cvc.py`
`test/integration/test_local_dataset.py`
 - Refactored test cases to work with new updates

[[JIRA Task](https://jiracommercial.flir.com/browse/CON-2763)]

### CODE REVIEW

File: docs/usage/cvc_guide.rst
Comment (rlc2) [inline docs/usage/cvc_guide.rst:286]: Missing a blank line between the '::' and the code snippet.

**Comment by rlc2** [comment]
From a UX perspective, it may be friendlier to users to at least keep around the `add` and `upload` commands, even if they only print a message about the new usage.  Also FYI, looking at the Conservator help page (at https://www.flirconservator.com/help#cvcOperationsForConservator), I see that it mentions the `conservator add` and `conservator upload` commands there.

**Review by rlc2 — APPROVED** [review]

**Comment by nbutler77** [comment]
@rlc2 Please don't merge this until it has a second review, ideally from @filitchp

**Comment by nbutler77** [comment]
@filitchp Can you or Alec give this a quick test drive?

**Comment by filitchp** [comment]
One minor initial comment is that the CLI help seems to cut off the description of the command very early so it reads like this:
```
Commands:
  add              The add command has been deprecated in favor of...
```
While it does read just fine when you do `cvc add --help` an alternative could be:
```
Commands:
  add              Deprecated. Please use stage-image instead.
```

**Comment by filitchp** [comment]
I noticed that `stage-image` by itself does not return anything. Should there be a message with some usage hints in that case?
```bash
$ cvc stage-image
$ cvc status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
No images staged.
```

**Comment by filitchp** [comment]
It's not possible to stage the original tiffs?
```bash
$ cvc stage-image analyticsData/video-DjiiGLnXpniWFj9ah-frame-000150-k3atrAjQSLGYYq2MM.tiff 
ERROR:FLIR.conservator.local_dataset:'analyticsData/video-DjiiGLnXpniWFj9ah-frame-000150-k3atrAjQSLGYYq2MM.tiff' is not a JPEG
```

File: FLIR/conservator/cli/cvc.py
Comment (filitchp) [inline FLIR/conservator/cli/cvc.py:48]: Thank you! That message has caused me to scratch my head many times...

File: FLIR/conservator/cli/cvc.py
Comment (filitchp) [inline FLIR/conservator/cli/cvc.py:114]: Typo: should be "configured"

**Comment by nbutler77** [comment]
> It's not possible to stage the original tiffs?
> 
> ```shell
> $ cvc stage-image analyticsData/video-DjiiGLnXpniWFj9ah-frame-000150-k3atrAjQSLGYYq2MM.tiff 
> ERROR:FLIR.conservator.local_dataset:'analyticsData/video-DjiiGLnXpniWFj9ah-frame-000150-k3atrAjQSLGYYq2MM.tiff' is not a JPEG
> ```

CVC has never supported image formats other than JPEG.

File: docs/usage/cvc_guide.rst
Comment (filitchp) [inline docs/usage/cvc_guide.rst:87]: This is some good documentation around the feature. One thing that's a little unclear to me is if you should ever stage your images from the `data/` directory. Would you do that with the `--skip-copy` flag or is it completely discouraged? Depending on the recommended behavior consider adding a warning with additional usage hints if the user tries to stage images directly from the current dataset's `data/`

**Review by filitchp — APPROVED** [review]
I ran an upload test on a dataset and functionally it works great. There is some usage confusion that I think could be addressed with warnings and usage hints. Otherwise looks good to me!

**Comment by nbutler77** [comment]
@filitchp If you have change requests, could you mark your review as "Changes Requested" instead of "Approved", please?

File: docs/usage/cvc_guide.rst
Comment (nbutler77) [reply docs/usage/cvc_guide.rst:87]: Honestly, I am not sure why the `--skip-copy` flag exists. When cvc uploads an image, it also generates an id for the new frame, and copies the image to the `data/` folder with a new name based on that new id, the dataset id, and other variables, so if you staged an image from `data/`, it would remain there, but would not affect anything else. if you use the `--skip-copy` flag, obviously the image would not get copied into `data/`, but I cannot think of a use-case where you would want to do that.

**Comment by nbutler77** [comment]
@filitchp Can you take another look at this PR, please?

File: docs/usage/cvc_guide.rst
Comment (filitchp) [reply docs/usage/cvc_guide.rst:87]: that makes sense, thanks Neal! Maybe we'll re-evaluate `--skip-copy` down the road, but it does not seem like it will hurt anything.

**Review by filitchp — APPROVED** [review]

**Comment by nbutler77** [comment]
@rlc2 @SJLC Since Paul approved, I think we're good to merge this PR.

**Pull request merged**
